### ISSUE TITLE
Withdraws do not work if wallet on external server server (Muhanmmed's way)

### ISSUE THREAD

**Issue by vetal20600** [peatio/peatio#667]
I've tried to apply suggested approach [here](https://github.com/muhammednagy/peatio/blob/master/doc/AWS.md) to keep wallet on another server. It is nice way to keep wallets on other server but I discovered that withdraws do not work.
At the same time internal wallets (and related source code) works fine.

```
RuntimeError (Circular dependency detected while autoloading constant Private::Withdraws::GamecreditsController):
  lib/middleware/security.rb:11:in `call'
```

**Comment by dkkoval** [comment]
Hi @vetal20600! Are you using `rpcuser` and `rpcpassword` in your *bitcoind* configuration?

If so, in *peatio* is your `rpc` in `config/currencies.yml` something like `"http://username:password@123.123.123.123:18332"`?

**Comment by vetal20600** [comment]
Yes, I set it properly. Deposit works well but I see error like above in 'production.log' while withdraws.

**Comment by vetal20600** [comment]
@dkkoval can you show your approach to run wallet on separate server ?

**Comment by dkkoval** [comment]
@vetal20600 I'm afraid right now I don't have anything to show you here. Peatio was designed to run on a single machine and it's a lot of work to upgrade apps like this. Soon we are making a huge upgrade of this part of Peatio. You can watch [our repo](https://github.com/rubykube/peatio)  in order not to miss latest news.

**Comment by muhammednagy** [comment]
hello @vetal20600  my way is working with more than 9 coins running on different servers 
can you give me more details?
thank you
